File: Python solutions/506. Relative Ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        pairs = []
        n = len(score)
        for i in range(n) :
            pair = (score[i], i)
            pairs.append(pair)
        
        pairs = sorted(pairs, key=lambda x: x[0], reverse=True)

        ans = [""] * n
        cnt = 1
        for pair in pairs :
            if cnt == 1 :
                ans[pair[1]] = "Gold Medal"
            elif cnt == 2 :
                ans[pair[1]] = "Silver Medal"
            elif cnt == 3 :
                ans[pair[1]] = "Bronze Medal"
            else :
                ans[pair[1]] = str(cnt)
            cnt += 1
        
        return ans

File: C++ solutions/506. Relative Ranks.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        vector<pair<int, int>> pairs;
        int n = score.size();
        for (int i = 0; i < n; ++i) {
            pairs.push_back({score[i], i});
        }

        sort(pairs.begin(), pairs.end(), greater<pair<int, int>>());

        vector<string> ans(n, "");

        int cnt = 1;
        for (auto& pair : pairs) {
            if (cnt == 1) {
                ans[pair.second] = "Gold Medal";
            } else if (cnt == 2) {
                ans[pair.second] = "Silver Medal";
            } else if (cnt == 3) {
                ans[pair.second] = "Bronze Medal";
            } else {
                ans[pair.second] = to_string(cnt);
            }
            cnt++;
        }

        return ans;
    }
};